code/modules/mob/living/simple_animal/hostile/human/nazi.dm

```diff
@@ -0,0 +1,372 @@
+			new /obj/item/ammo_storage/magazine/mc9mm/empty(loc)
+			ammo = 8
+		else
+			melee_damage_lower = 15
+			melee_damage_upper = 15
+			attack_sound = 'sound/weapons/bladeslice.ogg'
+			attacktext = "stabs"
+			retreat_distance = 0
+			minimum_distance = 0
+			ranged = 0
+
+/mob/living/simple_animal/hostile/humanoid/nazi/proc/droploot()
+	var/obj/item/weapon/gun/projectile/luger/dropgun = new(loc)
+	if(!ammo)
+		qdel(dropgun.chambered)
+		dropgun.chambered = null
+		qdel(dropgun.stored_magazine)
+	else
+		ammo--
+		var/removeAmmo = -1 * (ammo - 6)
+		for(var/i=1;i<=removeAmmo;i++)
+			var/obj/item/ammo_casing/C = pick(dropgun.stored_magazine.stored_ammo)
+			dropgun.stored_magazine.stored_ammo -= C
+			qdel(C)
+
+	for(var/i=1;i<=reloads;i++)
+		new /obj/item/ammo_storage/magazine/mc9mm(loc)
+
+///////////////////////////////////////////////////////////////////SOLDIER///////////
+
+/mob/living/simple_animal/hostile/humanoid/nazi/soldier
+	name = "Nazi Soldier"
+
+	icon_state = "nazisoldier"
+	icon_living = "nazisoldier"
+	icon_dead = "nazisoldier"
+
+	health = 150
+	maxHealth = 150
+
+	corpse = /obj/effect/landmark/corpse/nazi/soldier
+
+	melee_damage_lower = 5
+	melee_damage_upper = 5
+
+	retreat_distance = 4
+	minimum_distance = 3
+
+	projectiletype = /obj/item/projectile/energy/plasma/MP40k
+	projectilesound = 'sound/weapons/elecfire.ogg'
+
+	min_oxy = 0
+	max_oxy = 0
+	min_tox = 0
+	max_tox = 0
+	min_co2 = 0
+	max_co2 = 0
+	min_n2 = 0
+	max_n2 = 0
+
+	ammo = 13
+	ranged_cooldown_cap = 1
+	casingtype = null
+
+/mob/living/simple_animal/hostile/humanoid/nazi/soldier/Life()
+	..()
+	overlays.len = 0
+	if(stance != HOSTILE_STANCE_IDLE)
+		if(ranged)
+			overlays += icon('icons/mob/in-hand/right/guninhands_right.dmi',"PlasMP100")
+			if(melee_damage_upper > 10)
+				overlays += icon('icons/mob/in-hand/left/swords_axes.dmi',"knifenazi")
+		else
+			overlays += icon('icons/mob/in-hand/left/guninhands_left.dmi',"PlasMP0")
+			if (melee_damage_upper > 10)
+				overlays += icon('icons/mob/in-hand/right/swords_axes.dmi',"knifenazi")
+
+/mob/living/simple_animal/hostile/humanoid/nazi/soldier/afterShoot()
+	ammo--
+	if(ammo <= 0)
+		melee_damage_lower = 15
+		melee_damage_upper = 15
+		attack_sound = 'sound/weapons/bladeslice.ogg'
+		attacktext = "stabs"
+		retreat_distance = 0
+		minimum_distance = 0
+		ranged = 0
+		spawn(200)
+			if(health > 0)
+				ammo = 13
+				melee_damage_lower = 5
+				melee_damage_upper = 5
+				retreat_distance = 4
+				minimum_distance = 2
+				attacktext = "punches"
+				attack_sound = "punch"
+				ranged = 1
+
+/mob/living/simple_animal/hostile/humanoid/nazi/soldier/droploot()
+	var/obj/item/weapon/gun/energy/plasma/MP40k/dropgun = new(loc)
+	var/obj/item/weapon/cell/guncell = dropgun.power_supply
+	guncell.charge = min(guncell.maxcharge,25+(75 * ammo))
+
+
+///////////////////////////////////////////////////////////////////OFFICER///////////
+
+/mob/living/simple_animal/hostile/humanoid/nazi/officer
+	name = "Nazi Officer"
```

Comment:
Maybe add \improper in front of these capitalized names so it's "the greyshit stabs the Nazi Officer" rather than "the greyshit stabs Nazi Officer"?